src/components/TextNodeView.tsx

```diff
@@ -1,32 +1,38 @@
 import { Node } from "prosemirror-model";
+import { EditorState } from "prosemirror-state";
 import { Decoration, DecorationSet, EditorView } from "prosemirror-view";
 import { Component } from "react";
 import { findDOMNode } from "react-dom";
 
+import { reactKeysPluginKey } from "../plugins/reactKeys.js";
 import { CompositionViewDesc, TextViewDesc, ViewDesc } from "../viewdesc.js";
 
 import { wrapInDeco } from "./ChildNodeViews.js";
 
 type Props = {
   view: EditorView | null;
+  state: EditorState | null;
   node: Node;
   pos: number;
-  siblingDescriptors: ViewDesc[];
+  viewDescContext: Record<string, ViewDesc>;
   decorations: readonly Decoration[];
 };
 
 export class TextNodeView extends Component<Props> {
   private viewDescRef: null | TextViewDesc = null;
 
   updateEffect() {
-    const { decorations, siblingDescriptors, node } = this.props;
+    const { decorations, viewDescContext, node, state, pos } = this.props;
+    const reactKeysState = state && reactKeysPluginKey.getState(state);
+    const key = reactKeysState?.posToKey.get(pos);
+    const parentKey = key && reactKeysState?.keyToParent.get(key);
+    const parent =
+      parentKey !== undefined ? viewDescContext[parentKey] : undefined;
+
```

Comment:
This is the actual fix to #128: when Strict Mode re-renders just the TextNodeView, we are still able to use the key map to find the parent view descriptor, so we can maintain the bidirectionality of the view descriptor tree